### PR TITLE
[fsdp] fix: FSDP logits temperature scaling for view tensors

### PR DESCRIPTION
**Pull request by zjchenn**
### What does this PR do?

This PR updates the FSDP language-model eager logits path to avoid in-place temperature scaling on a squeezed logits view.

`output.logits.squeeze(0)` can return a view. For models whose logits are produced through custom autograd functions, applying `div_()` to that view can be rejected by PyTorch because it may interfere with the custom backward path.

The change also mirrors the automodel path by materializing DTensor logits with `full_tensor()` before log-probability computation when logits are sharded on the vocab dimension.

Related implementations:

torchtitan
```python
labels = labels.squeeze(0)
logits_rmpad = logits.squeeze(0)
# PyTorch's autograd doesn't allow in-place modification of views when gradients need to flow back
logits_rmpad = logits_rmpad / temperature
```

automodel
```python
logits_rmpad = output.logits.squeeze(0)
# With TP, logits are DTensors sharded on vocab dim; gather for log_softmax.
if isinstance(logits_rmpad, DTensor):
    logits_rmpad = logits_rmpad.full_tensor()
logits_rmpad = logits_rmpad / temperature_rmpad.clamp(min=1e-8).unsqueeze(-1).to(logits_rmpad.dtype)
```

### Checklist Before Starting

- [x] Search for similar PRs. Paste at least one query link here: ...
- [x] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`, `fully_async`, `one_step_off`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

> For changes that can not be tested by CI (e.g., algorithm implementation, new model support), validate by experiment(s) and show results like training curve plots, evaluation results, etc.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [x] Read the [Contribute Guide](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md).
- [x] Apply [pre-commit checks](https://github.com/verl-project/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/verl-project/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/verl-project/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

File: verl/workers/engine/fsdp/transformer_impl.py [hunk 1]
```diff
@@ -1136,7 +1136,10 @@ def prepare_model_outputs(self, output, output_args, micro_batch: TensorDict, lo
                             model_output[field_name] = torch.nested.nested_tensor_from_jagged(v, cu_seqlens)
             else:
                 logits_rmpad = output.logits.squeeze(0)  # (total_nnz, vocab_size)
-                logits_rmpad.div_(temperature_rmpad.clamp(min=1e-8).unsqueeze(-1).to(logits_rmpad.dtype))
```
Comment (wuxibin89) [inline verl/workers/engine/fsdp/transformer_impl.py:1139]: The BSHD branch should adjust as well:
https://github.com/verl-project/verl/blob/main/verl/workers/engine/fsdp/transformer_impl.py#L1249

Comment (zjchenn) [reply verl/workers/engine/fsdp/transformer_impl.py:1139]: Done, thanks for the reminder. I’ve updated the BSHD branch.

**Review by wuxibin89 — APPROVED** [review]
